src/webui/src/App.tsx

```diff
@@ -99,14 +102,22 @@ class App extends React.Component<{}, AppState> {
         this.setState({ bestTrialEntries: entries });
     }
 
+    shouldComponentUpdate(nextProps: any, nextState: AppState): boolean {
```

Comment:
Don't do this...

Reply:
If don't use shouldComponentUpdate, parent App.tsx component will refresh three times when experiment done, and then child components will also refresh.  And App.tsx's shouldComponentUdate only watch a boolean value.